library/src/main/java/com/novoda/downloadmanager/lib/Request.java

```diff
@@ -234,6 +235,17 @@ public Request alwaysAttemptResume() {
     }
 
     /**
+     * Automatically pause the download when reaching the end of the file instead of writing the last bytes to disk.
+     * This allows for a resume of the download against an updated tar file.
+     * If used this also sets {@code alwaysAttemptResume()}
+     */
+    public Request allowTarUpdates() {
+        allowTarUpdates = true;
+        alwaysAttemptResume();
```

Comment:
:+1: If Ferran is already working on this, I'd prefer this gets merged in - especially as the API won't change.